### ISSUE TITLE
opt_stop_early not working properly

### ISSUE THREAD

**Issue by lucas-johnson** [Jean-Romain/lidR#308]
### Current behavior
The `opt_stop_early` setting doesn't effect the `LAScatalog` processing:

```r
opt_stop_early(testcat) = FALSE
lasnormalize(testcat, knnidw(k=5))
#> Processing [==============================================>-----------------------------------------------]  50% (1/2) eta:  1s
#> Error in lasnormalize.LAS(x, algorithm, na.rm, use_class, ...) : 
#> No ground point found in the point cloud.
opt_stop_early(testcat) = TRUE
lasnormalize(testcat, knnidw(k=5))
#> Processing [==============================================>-----------------------------------------------]  50% (1/2) eta:  1s
#> Error: No ground point found in the point cloud.
```

### Desired behavior
When `opt_stop_early(ctg)` is set to `FALSE` then the processing of the catalog should continue despite encountering an error with one of the chunks.

**Comment by Jean-Romain** [comment]
Reproducible example

```r
library(lidR)
ctg = readLAScatalog("~/Documents/ALS data/Haliburton dataset/Landscape LiDAR/")

lidR:::catalog_fakerun(ctg, pwarning = 0, perror = 0.1)

opt_stop_early(ctg) <- FALSE
lidR:::catalog_fakerun(ctg, pwarning = 0, perror = 0.1)
````

**Comment by Jean-Romain** [comment]
Fixed in v2.1.5
